### PR TITLE
Review delete

### PR DESCRIPTION
**Pull request by acstewart724**

### CODE REVIEW

**Comment by acstewart724** [comment]
finish deleteReview function
edit reviewRouter res.status

Co-authored-by: camziny <camziny@users.noreply.github.com>
Co-authored-by: Fabricius Eusebio <yoFabricius@users.noreply.github.com>
Co-authored-by: Tyler <br.ndt.tyler@gmail.com>

File: server/src/routes/api/v1/reviewsRouter.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import express from 'express'
+import { Review } from "../../../models/index.js"
+
+const reviewsRouter = new express.Router()
+
+reviewsRouter.delete("/:id", async (req, res) => {
+  try {
+    await Review.query().deleteById(req.params.id)
```
Comment (dovakeen118) [inline server/src/routes/api/v1/reviewsRouter.js]: We wouldn't want someone to be able to delete someone else's review, and can check for that in the router here to see if the current user (`req.user`) is the same user that created this review to be deleted

File: client/src/components/ReviewTile.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
-const ReviewTile = ({ title, rating, content }) => {
+const ReviewTile = ({ id, title, rating, content, deleteReview }) => {
+
   return (
     <div>
       <h4>{title}</h4>
       <h5>{rating}</h5>
       <p>{content}</p>
+      <div onClick={() => { deleteReview(id) }}>
+        <input type="submit" value="Delete Review"/>
```
Comment (dovakeen118) [inline client/src/components/ReviewTile.js]: we could instead make this a button and move the `onClick` to that element as well - I'd say though the key piece is putting the synthetic click event as close to the actual thing that's being interacted with

Comment (dovakeen118) [inline client/src/components/ReviewTile.js]: you might also consider only making this delete visible if the signed in user is the same one who created the review

**Review by dovakeen118 — COMMENTED** [review]
Nice work! there's a couple things to implement to fully complete the feature, and a few recommendations

File: client/src/components/AttractionShowPage.js [hunk 3]
```diff
@@ -31,12 +32,42 @@ const AttractionShowPage = (props) => {
...
+          return review.id !== reviewId
+        })
+        setErrors([])
+        setAttraction({...attraction, reviews: filteredReviews})
+      }
+      } catch (error) {
+          console.error(`Error in fetch: ${error.message}`)
+      }
+      }
+
```
Comment (nwalberts) [inline client/src/components/AttractionShowPage.js:61]: unfortunately I have to b e a stickler. the indentation is off!

File: client/src/components/AttractionShowPage.js [hunk 4]
```diff
@@ -31,12 +32,42 @@ const AttractionShowPage = (props) => {
...
+        if (response.status === 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        }
+      } else {
```
Comment (nwalberts) [inline client/src/components/AttractionShowPage.js:49]: you control for 422, but your router isnt setup to respond with a 422 here. Instead, please prepare for a 401 error

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**